### PR TITLE
PLT-1346: Introduce Basic OCI

### PR DESCRIPTION
**Pull request by wcrum**
Current behavior is below, where `func() resourceRegistryOciEcr` schema handles both ECR and Basic repositories, if needed I can have separation between basic and ecr, but this would require changes to the base resource spectrocloud_registry_oci.

```terraform
resource "spectrocloud_registry_oci" "r1" {
  name       = "test-nik4"
  type       = "basic" # basic
  endpoint   = "harbor.registry"
  provider_type = "helm"
  credentials {
    credential_type = "basic"
    username = "username"
    password = "password"
  }
}
```

Additionally, given that OCI registries exist at a tenant level, would need provider.go to support tenant level actions, like the commented `client.WithScopeTenant()(c)`

https://github.com/spectrocloud/terraform-provider-spectrocloud/blob/d113161b775cd92fff47662eb72cc1aaef244f40/spectrocloud/provider.go#L240-L242

### CODE REVIEW

File: spectrocloud/resource_registry_oci_ecr.go [hunk 1]
```diff
@@ -46,6 +49,10 @@ func resourceRegistryOciEcr() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"provider_type": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (SivaanandM) [inline spectrocloud/resource_registry_oci_ecr.go:54]: Can we make this as optional and set as "helm" to default as similar to UI. if we made it as required. if customer bump up the tf version they need to update their configurations

**Comment by SivaanandM** [comment]
> Current behavior is below, where `func() resourceRegistryOciEcr` schema handles both ECR and Basic repositories, if needed I can have separation between basic and ecr, but this would require changes to the base resource spectrocloud_registry_oci.
> 
> ```hcl
> resource "spectrocloud_registry_oci" "r1" {
>   name       = "test-nik4"
>   type       = "basic" # basic
>   endpoint   = "harbor.registry"
>   provider_type = "helm"
>   credentials {
>     credential_type = "basic"
>     username = "username"
>     password = "password"
>   }
> }
> ```
> 
> Additionally, given that OCI registries exist at a tenant level, would need provider.go to support tenant level actions, like the commented `client.WithScopeTenant()(c)`
> 
> https://github.com/spectrocloud/terraform-provider-spectrocloud/blob/d113161b775cd92fff47662eb72cc1aaef244f40/spectrocloud/provider.go#L240-L242

So basically registries can be created only and tenant level. and used across the projects. this is handled in resource level itself. not required to update provider.go

**Comment by SivaanandM** [comment]
Kindly address the review comments. We will prioritize the changes for testing and include them in the corresponding release.

**Review by SivaanandM — CHANGES_REQUESTED** [review]

File: spectrocloud/resource_registry_oci_ecr.go (hunk 1 above)
Comment (wcrum) [reply spectrocloud/resource_registry_oci_ecr.go:54]: Done.

**Review by SivaanandM — APPROVED** [review]

**Pull request merged**